f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py

```diff
@@ -167,26 +168,102 @@ def __init__(self, driver):
         self._resource = "virtual address"
         self.resource_helper = resource_helper.BigIPResourceHelper(
             resource_helper.ResourceType.virtual_address)
+        self.tenant_manager = tenants.BigipTenantManager(self.driver.conf,
+                                                         self.driver)
         self.mutable_props = {
             "name": "description",
             "description": "description",
             "admin_state_up": "enabled"
         }
+        self.all_subnet_hints = {}
 
-    def _create_payload(self, loadbalancer, service):
+    def _create_vip(self, loadbalancer, service):
         vip = virtual_address.VirtualAddress(self.driver.service_adapter,
                                              loadbalancer)
-        return vip.model()
+        return vip
 
     @log_helpers.log_method_call
     def create(self, loadbalancer, service, **kwargs):
-        # TODO(qzhao): Future work
-        pass
+        self._pre_create(service)
+        self._create_lb(service)
+        self._post_create(service)
+
+    def _post_create(self, service):
+        # create fdb for vxlan tunnel
+        if not self.driver.conf.f5_global_routed_mode:
+            self.driver.network_builder.update_bigip_l2(service)
+
+    def _create_lb(self, service):
+        bigips = self.driver.get_config_bigips()
+        loadbalancer = service["loadbalancer"]
+        vip = self._create_vip(loadbalancer, service)
+
+        for bigip in bigips:
+            vip.assure(bigip)
+
+        # allow address pair
+        if self.driver.l3_binding:
```

Comment:
If it is something that need to be done after all vip are created, you may move it to _post_create()

Reply:
> If it is something that need to be done after all vip are created, you may move it to _post_create()

value of vip_address has already existed, this needs some scenario to confirm if it's necessary to be created before vip.